CMakeLists.txt

```diff
@@ -589,21 +600,34 @@ add_custom_target(update_source_tree
 )
    
 
-   
 #### create rev_num.h - file with some info about code origin
 # a custom target - is always built -> call cmake script to generate rev_num.h
 # Get the current working branch
 
 # This command is run from build dir and cmake set CMAKE_SOURCE_DIR to the build dir; thus we have to 
 # pass different variable with source dir, namely FLOW123D_SOURCE_DIR
+set(CONFIG_FILE_REV_NUM "${CONFIG_FILE_INCLUDES_DIRECTORY}/rev_num.h")
 add_custom_target(rev_num_always 
-    COMMAND ${CMAKE_COMMAND}  -DFLOW123D_SOURCE_DIR=${CMAKE_SOURCE_DIR} -P ${CMAKE_SOURCE_DIR}/CMake/Modules/Flow123d_git_rev_num.cmake 
+    COMMAND ${CMAKE_COMMAND}  -DFLOW123D_SOURCE_DIR=${CMAKE_SOURCE_DIR} -DFILE_PATH=${CONFIG_FILE_REV_NUM} -P ${CMAKE_SOURCE_DIR}/CMake/Modules/Flow123d_git_rev_num.cmake 
 )
 # rev_num.h is a generated file
-set_source_files_properties(${CMAKE_BINARY_DIR}/rev_num.h
+set_source_files_properties(${CONFIG_FILE_REV_NUM}
+    PROPERTIES GENERATED TRUE
+    HEADER_FILE_ONLY TRUE)
+
+### create config.h - file containing definitions flags and other information
+# called everytime
+set(CONFIG_FILE_HEADER "${CONFIG_FILE_INCLUDES_DIRECTORY}/config.h")
+GENERATE_CONFIG_H ("${CONFIG_FILE_HEADER}.tmp")
+add_custom_target(config_h_always 
```

Comment:
If we have split configure time and build time defines, possibly we need not to use config_h_always target to update the config.h since sonfig.h.tmp should not change.